tracing-attributes/tests/parents.rs

```diff
@@ -18,26 +18,15 @@ fn default_parent_test() {
     let child = expect::span().named("with_default_parent");
 
     let (collector, handle) = collector::mock()
-        .new_span(
-            contextual_parent
-                .clone()
-                .with_contextual_parent(None)
-                .with_explicit_parent(None),
```

Comment:
it seems like the fact that both of these methods can be called on an `ExpectedSpan` creates some ambiguity in the `tracing-mock` API surface. what do you think about getting rid of `with_contextual_parent()` and `with_explicit_parent()`, and replacing them with a single `with_parent()` method taking a `Parent` enum? something like this:
```rust
expect::span()
   .with_parent(Parent::Contextual(Some("whatever"))
```
this way, the user can no longer construct a span that expects both a contextual parent and an explicit parent at the same time.

Reply:
I started doing this, but the change is huge. It also has some unfortunate side-efffects (at lesat the way that I implemented it). Because then we'll need:

```rust
expect::span()
    .with_parent(Parent::Contextual(Some("whatever".to_owned())))
```

Which starts to get a bit long. Also, exposing the `Parent` enum makes everything longer too.

I think that there is room for improvement here, but I would like to look at it in a separate PR so as not to make this one too large. What do you think?